xilem_core/src/views/orphan.rs

```diff
@@ -0,0 +1,125 @@
+/// Every type that can be supported with this trait, needs a concrete `View` implementation in xilem_core, possibly feature-gated.
+pub trait OrphanView<V, State, Action, Message = DynMessage>: ViewPathTracker + Sized {
+    /// See [`View::Element`]
+    type OrphanElement: ViewElement;
+    /// See [`View::ViewState`]
+    type OrphanViewState;
+
+    /// See [`View::build`]
+    fn orphan_build(view: &V, ctx: &mut Self) -> (Self::OrphanElement, Self::OrphanViewState);
+
+    /// See [`View::rebuild`]
+    fn orphan_rebuild<'el>(
+        new: &V,
+        prev: &V,
+        view_state: &mut Self::OrphanViewState,
+        ctx: &mut Self,
+        element: Mut<'el, Self::OrphanElement>,
+    ) -> Mut<'el, Self::OrphanElement>;
+
+    /// See [`View::teardown`]
+    fn orphan_teardown(
+        view: &V,
+        view_state: &mut Self::OrphanViewState,
+        ctx: &mut Self,
+        element: Mut<'_, Self::OrphanElement>,
+    );
+
+    /// See [`View::message`]
+    fn orphan_message(
+        view: &V,
+        view_state: &mut Self::OrphanViewState,
+        id_path: &[ViewId],
+        message: Message,
+        app_state: &mut State,
+    ) -> MessageResult<Action, Message>;
+}
+
+macro_rules! impl_orphan_view_for {
+    ($ty: ty) => {
+        impl<State, Action, Context, Message> View<State, Action, Context, Message> for $ty
+        where
+            Context: OrphanView<$ty, State, Action, Message>,
+        {
+            type Element = Context::OrphanElement;
+
+            type ViewState = Context::OrphanViewState;
+
+            fn build(&self, ctx: &mut Context) -> (Self::Element, Self::ViewState) {
+                Context::orphan_build(self, ctx)
+            }
+
+            fn rebuild<'el>(
+                &self,
+                prev: &Self,
+                view_state: &mut Self::ViewState,
+                ctx: &mut Context,
+                element: Mut<'el, Self::Element>,
+            ) -> Mut<'el, Self::Element> {
+                Context::orphan_rebuild(self, prev, view_state, ctx, element)
+            }
+
+            fn teardown(
+                &self,
+                view_state: &mut Self::ViewState,
+                ctx: &mut Context,
+                element: Mut<'_, Self::Element>,
+            ) {
+                Context::orphan_teardown(self, view_state, ctx, element);
+            }
+
+            fn message(
+                &self,
+                view_state: &mut Self::ViewState,
+                id_path: &[ViewId],
+                message: Message,
+                app_state: &mut State,
+            ) -> MessageResult<Action, Message> {
+                Context::orphan_message(self, view_state, id_path, message, app_state)
+            }
+        }
+    };
+}
+
+// string impls
+impl_orphan_view_for!(&'static str);
+impl_orphan_view_for!(alloc::string::String);
+impl_orphan_view_for!(alloc::borrow::Cow<'static, str>);
+
+// number impls
```

Comment:
I'm still not convinced that these are that meaningful, but I'm not going to block on that.

Reply:
Yeah apart from a little bit macro-expanded boilerplate (which I don't think should contribute too much to the compilation) doesn't hurt?
(In case we can also feature gate this, like the kurbo module I guess, would probably be interesting whether this meaningfully contributes to compilation time...)

I think it's cool to have something like (taken from xilem_web) `p(("The answer to life is ", 42))` and it should be cheap as view, since equality checks are trivial.